.github/workflows/java_nightly.yml

```diff
@@ -78,6 +103,12 @@ jobs:
             cp binaries/$PREFIX/java-jars/$LIBRARY repo/org/apache/arrow/${BASH_REMATCH[1]}/${BASH_REMATCH[2]}
             echo "Artifacts $LIBRARY configured"
           done
+      - name: Prune Repository
+        shell: bash
+        env:
+          KEEP: ${{ github.event.inputs.keep || 14 }}
+        run: |
+          for i in `ls -t repo/org/apache/arrow`; do; find repo/org/apache/arrow/$i -mindepth 1 -maxdepth 1 -type d -print0 | xargs -0 ls -t -d | tail -n +$((KEEP + 1)) | xargs rm -rf; done
       - name: Show repo contents
         run: tree repo
       - name: Upload Files
```

Comment:
You also need to update this step with the new action and switches otherwise the locally deleted files wont be deleted from the server.

Reply:
Thank you @assignUser , just added